heartbeat/gcp-vpc-move-ip

```diff
@@ -0,0 +1,302 @@
+  while request is not None:
+    response = request.execute()
+    zones = response.get('items', {})
+    for zone in zones.values():
+      for inst in zone.get('instances', []):
+        if inst['name'] == instance:
+          return inst['zone'].split("/")[-1]
+    request = CONN.instances().aggregatedList_next(
+        previous_request=request, previous_response=response)
+  raise Exception("Unable to find instance %s" % (instance))
+
+
+def gcp_alias_start(alias):
+  if not alias:
+    sys.exit(OCF_ERR_CONFIGURED)
+  my_alias = get_localhost_alias()
+  my_zone = get_metadata('instance/zone').split('/')[-1]
+  project = get_metadata('project/project-id')
+
+  # If I already have the IP, exit. If it has an alias IP that isn't the VIP,
+  # then remove it
+  if my_alias == alias:
+    LOGGER.info(
+        '%s already has %s attached. No action required' % (HOSTNAME, alias))
+    sys.exit(OCF_SUCCESS)
+  elif my_alias:
+    LOGGER.info('Removing %s from %s' % (my_alias, HOSTNAME))
+    set_alias(project, my_zone, HOSTNAME, '')
+
+  # Loops through all hosts & remove the alias IP from the host that has it
+  hostlist = os.environ.get('OCF_RESKEY_hostlist', '')
+  hostlist.replace(HOSTNAME, '')
+  for host in hostlist.split():
+    host_zone = get_zone(project, host)
+    host_alias = get_alias(project, host_zone, host)
+    if alias == host_alias:
+      LOGGER.info(
+          '%s is attached to %s - Removing all alias IP addresses from %s' %
+          (alias, host, host))
+      set_alias(project, host_zone, host, '')
+      break
+
+  # add alias IP to localhost
+  set_alias(
+      project, my_zone, HOSTNAME, alias,
+      os.environ.get('OCF_RESKEY_alias_range_name'))
+
+  # Check the IP has been added
+  my_alias = get_localhost_alias()
+  if alias == my_alias:
+    LOGGER.info('Finished adding %s to %s' % (alias, HOSTNAME))
+  elif my_alias:
+    LOGGER.error(
+        'Failed to add IP. %s has an IP attached but it isn\'t %s' %
+        (HOSTNAME, alias))
+    sys.exit(OCF_ERR_GENERIC)
+  else:
+    LOGGER.error('Failed to add IP address %s to %s' % (alias, HOSTNAME))
+    sys.exit(OCF_ERR_GENERIC)
+
+
+def gcp_alias_stop(alias):
+  if not alias:
+    sys.exit(OCF_ERR_CONFIGURED)
+  my_alias = get_localhost_alias()
+  my_zone = get_metadata('instance/zone').split('/')[-1]
+  project = get_metadata('project/project-id')
+
+  if my_alias == alias:
+    LOGGER.info('Removing %s from %s' % (my_alias, HOSTNAME))
+    set_alias(project, my_zone, HOSTNAME, '')
+
+
+def gcp_alias_status(alias):
+  if not alias:
+    sys.exit(OCF_ERR_CONFIGURED)
+  my_alias = get_localhost_alias()
+  if alias == my_alias:
+    LOGGER.info('%s has the correct IP address attached' % HOSTNAME)
+  else:
+    sys.exit(OCF_NOT_RUNNING)
+
+
+def configure():
+  global CONN
+  global LOGGER
+  global HOSTNAME
+
+  # Populate global vars
+  CONN = googleapiclient.discovery.build('compute', 'v1')
+  HOSTNAME = platform.node()
```

Comment:
This should be "get_metadata('instance/name')" as hostname isnt always the same as the instance name.

Reply:
hostname should be the name of the machine (it is in the test requirements at https://github.com/GoogleCloudPlatform/compute-image-tools/blame/master/daisy_workflows/image_test/configuration/README.md#L10) but I agree it is not reliable, I'll update the PR with this change.